MSAL/src/util/ios/MSALRedirectUriVerifier.m

```diff
@@ -26,63 +26,111 @@
 //------------------------------------------------------------------------------
 
 #import "MSALRedirectUriVerifier.h"
+#import "MSALRedirectUri+Internal.h"
+#import "MSIDConstants.h"
 
 @implementation MSALRedirectUriVerifier
 
-+ (BOOL)verifyRedirectUri:(NSURL *)redirectUri
-            brokerEnabled:(BOOL)brokerEnabled
-                    error:(NSError **)error
++ (MSALRedirectUri *)msalRedirectUriWithCustomUri:(NSString *)customRedirectUri
+                                         clientId:(NSString *)clientId
+                                            error:(NSError * __autoreleasing *)error
 {
-    NSString *scheme = redirectUri.scheme;
-
-    if (![self verifySchemeIsRegistered:scheme error:error])
+#if AD_BROKER
+    // Allow the broker app to use a special redirect URI when acquiring tokens
+    if ([url isEqualToString:MSID_AUTHENTICATOR_REDIRECT_URI])
     {
-        return NO;
+        return YES;
```

Comment:
What does this indicate?
Should we return MSALRedirectUri created with authenticator_redirect_uri?

Reply:
Of course it should :) Fixed.